playwright_tests/E2E/steps/staff-steps.ts

```diff
@@ -1,15 +1,17 @@
-  export async function clickToStaffPage(page) {
-    console.log("Going to Staff Page");
-    const numberOfTries = 5;
-    for (let i = 0; i < numberOfTries; i++) {
-      try {
-        await page.getByRole('link', { name: 'Staff' }).click();
-        await page.waitForSelector('text=User list', { state: 'visible', timeout: 1000 });
-        break;
-      } catch (error) {
-        console.log(`Click to staff page failed, retrying (${i + 1}/${numberOfTries})...`);
-        if (i !== numberOfTries) await page.waitForTimeout(1000); // Wait 1 second before the next attempt
-        else throw error;
-      }
-    }
-  }
\ No newline at end of file
+import  { retryAction } from './retry-steps';
+
+export async function clickToStaffPage(page) {
+  console.log("Going to Staff Page");
+  await retryAction(async () => {
+    await page.getByRole('link', { name: 'Staff' }).click();
+    await page.waitForSelector('text=User list', { state: 'visible', timeout: 1000 });
+  });
+}
+
+export async function fillSearchBox(page) {
```

Comment:
I'd be happier if this function had an argument for the text to submit, perhaps with a default, e.g. function fillSearchBox(page, text='xui').